File: Data Structures & Algorithms/contains-duplicate-ii/submission-3.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        dist = {}

        for i in range(len(nums)):
            if nums[i] in dist and i - dist[nums[i]] <= k:
                return True

            dist[nums[i]] = i
        
        return False
        

File: Data Structures & Algorithms/majority-element/submission-3.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums: number[]): number {
        let count: number = 0;
        let candidate = null;

        for (let i: number = 0; i < nums.length; i++) {
            if (count === 0) {
                candidate = nums[i];
            }

            if (nums[i] === candidate) {
                count++;
            } else {
                count--;
            }
        }

        return candidate
    }
}
